Makefile.dep

```diff
@@ -206,6 +206,9 @@ ifneq (,$(filter gnrc_netif,$(USEMODULE)))
   ifneq (,$(filter gnrc_lorawan,$(USEMODULE)))
     USEMODULE += gnrc_netif_lorawan
   endif
+  ifneq (,$(filter netif_pktq,$(USEMODULE)))
```

Comment:
Not sure how this would be expressed in the build system: but shouldn't `netif_pktq` rather be a feature that the driver may request? If a stack does not support this for whatever reason you would still be able to use it without `netif_pktq` and the user wouldn't be bothered with such a low-level decision.

Reply:
Are you suggesting something like `FEATURE_OPTIONAL`? 
IIRC modules can't provide features with the current build system.

Or should I just add `USEMODULE += netif_pktq` to the driver's `Makefile.dep` and drop the blocking code entirely?

Reply:
I meant the first, but did not know, that it is not easily realizable, currently. I would not enforce the packet queue, as one might want to optimize it out.

Reply:
> I meant the first, but did not know, that it is not easily realizable, currently. I would not enforce the packet queue, as one might want to optimize it out.

For now `FEATURES_OPTIONAL` does not apply to this use case. The only thing that kind of resembles this is `DEFAULT_MODULE` but that has its own issues, would only work if `gnrc_netif_pktq` has no dependencies of its own.

AFAIK this seems like the right way for the use to include `netif_pktq` if solicited, its an `opt-in` policy, the only `opt-out` option for modules is `DEFAULT_MODULE`.

Reply:
@miri64 I think the way it is modeled currently is OK considered our current buildystem, should we go ahead with this one? Are there any other showstoppers?

Reply:
Mhh after thinking about this a bit: what @benpicco actually want's to express to pull in the queue is "the device does not support blocking send so the network stack requires a packet queue" not "the netif has a packet queue so the network stack needs a packet queue". Is that expressible in the build system? In any case, I would prefer a move to a semantic of requiring `netif_pktq` to include this to a feature `netdev_blocking_send` by the device or something that is similar.

Reply:
Alternatively, this should be a pseudo-submodule of the `at86rf215` driver: `at86rf215_blocking_send` and the `gnrc_netif_pktq` module included as a dependency of that (maybe via a generalized pseudo-module `netdev_blocking_send`).

Reply:
Ping @benpicco?

Reply:
So many options which one should I take? :smile:

Reply:
I would prefer the last one :-).

Reply:
Pulling in `at86rf215_blocking_send` when `gnrc_netif_pktq` is *not* selected sounds like the best aproach.

Reply:
The only problem I see is that now drivers can't depend on having a `netif_pktq` available and *still* have to implement a fall-back blocking send :\

Reply:
That will be addressed by @jia200x's new API though, right?